File: looophole/CPP/477.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int sum = 0;
        int l = nums.size();
        for(int i = 0; i<l-1;i++){
            for(int j=i+1; j<l;j++){
                int a = nums[i]^nums[j];
                int b = __builtin_popcount(a);
                sum = sum + b;
            }
        }
        return sum;
    }
};

File: looophole/Python/477.py
class Solution:
    def totalHammingDistance(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        for bit in range(32):
            count_ones = 0
            for num in nums:
                if num & (1 << bit):
                    count_ones += 1
            count_zeros = n - count_ones
            ans += count_ones * count_zeros
            
        return ans
